golos.vesting/golos.vesting.cpp

```diff
@@ -541,5 +545,32 @@ void vesting::timeout_delay_trx() {
     trx.send(_self.value, _self);
 }
 
+void vesting::paydelegator(name voter, asset reward, name delegator, 
+        uint64_t interest_rate, uint8_t payout_strategy) {
+    if (payout_strategy == config::to_delegated_vesting) {
+        tables::account_table acc_table_dlg(_self, delegator.value);
+        auto balance_dlg = acc_table_dlg.find(reward.symbol.code().raw());
+        acc_table_dlg.modify(balance_dlg, voter, [&](auto& item) {
+            item.delegate_vesting += reward;
+        });
+        tables::account_table acc_table_rcv(_self, voter.value);
+        auto balance_rcv = acc_table_rcv.find(reward.symbol.code().raw());
+        acc_table_rcv.modify(balance_rcv, delegator, [&](auto& item) {
+            item.received_vesting += reward;
+        });
+        tables::delegate_table table(_self, reward.symbol.code().raw());
+        auto index_table = table.get_index<"unique"_n>();
+        auto delegate_record = index_table.find(structures::delegate_record::unique_key(delegator, voter));
```

Comment:
here can be edge case, when delegation object is no more exists at payout time (delegator returned his delegation after delegatee upvote). in this case whole this block is skipped, simple increasing delegator's vesting balance is enough. please refer Golos code: https://github.com/GolosChain/golos/blob/9b2d6d6f8f4fdebb19eda83e766eef13695390b0/libraries/chain/database.cpp#L2304-L2319